src/Common/src/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -59,6 +59,22 @@ public static bool IsBlittableType(TypeDesc type)
             return false;
         }
 
+        public static bool IsStructMarshallingRequired(TypeDesc typeDesc)
+        {
+            if (typeDesc.Category != TypeFlags.ValueType)
+                return false;
+
+
+            var type = typeDesc.UnderlyingType as MetadataType;
+            
+            // if it is auto layout we don't need marshalling because
+            // the marshaller will throw anyways
+            if (!type.IsSequentialLayout && !type.IsExplicitLayout)
+                return false;
```

Comment:
I suspect you need to say "it's required" and then throw in the right thunks. Otherwise you could easily end up marshalling them anyway.